src/components/pages/Registration/index.tsx

```diff
@@ -0,0 +1,135 @@
+/********************************************************************************
+ * Copyright (c) 2021, 2023 BMW Group AG
+ * Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+import { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Button, Stepper, Typography } from '@catena-x/portal-shared-components'
+import './Registration.scss'
+import {
+  ApplicationStatus,
+  useFetchApplicationsQuery,
+} from 'features/registration/registrationApiSlice'
+
+export default function Registration() {
+  const { t } = useTranslation()
+
+  const [activeStep, setActiveStep] = useState(1)
+
+  const { data } = useFetchApplicationsQuery()
+  const companyData = data?.[0]
+
+  useEffect(() => {
+    if (
+      companyData?.applicationStatus === ApplicationStatus.CREATED ||
+      companyData?.applicationStatus === ApplicationStatus.ADD_COMPANY_DATA
+    ) {
+      setActiveStep(1)
+    } else if (
+      companyData?.applicationStatus === ApplicationStatus.INVITE_USER
+    ) {
+      setActiveStep(2)
+    } else if (
+      companyData?.applicationStatus === ApplicationStatus.SELECT_COMPANY_ROLE
+    ) {
+      setActiveStep(3)
+    } else if (
+      companyData?.applicationStatus === ApplicationStatus.UPLOAD_DOCUMENTS
+    ) {
+      setActiveStep(4)
+    } else if (companyData?.applicationStatus === ApplicationStatus.VERIFY) {
+      setActiveStep(5)
+    }
```

Comment:
please use a switch statement instead of this long chain of if-else

Reply:
@oyo Made the change. Please review now